flow/projecttracking/jira/jira.py

```diff
@@ -0,0 +1,376 @@
+                project_detail.get('url', '')), 'ERROR')
+            commons.print_msg(Jira.clazz, method, e, 'ERROR')
+            exit(1)
+
+        if resp.status_code == 200:
+            json_data = json.loads(resp.text)
+            commons.print_msg(Jira.clazz, method, "Project Key: {key}, Project Id: {id}".format(key=json_data['key'], 
+                                                                                                id=json_data['id']))
+        else:
+            commons.print_msg(Jira.clazz, method, "Failed retrieving project detail from call to {url}. \r\n "
+                                                "Response: {response}".format(url=story_detail.get('url', ''),
+                                                                            response=resp.text), 'WARN')
+
+        commons.print_msg(Jira.clazz, method, 'end')
+        return json_data
+
+
+    def _retrieve_story_detail(self, story_id):
+        method = '_retrieve_story_detail'
+        commons.print_msg(Jira.clazz, method, 'begin')
+
+        json_data = None
+        resp = None
+
+        story_detail = {'url': '{0}/rest/api/3/issue/{1}'.format(Jira.jira_url, story_id)}
+        
+        headers = {'Content-type': 'application/json', 'Accept': 'application/json', 'Authorization': 'Basic {0}'.format(Jira.jira_basic_auth)}
+
+        try:
+            commons.print_msg(Jira.clazz, method, story_detail['url'])
+            resp = requests.get(story_detail['url'], headers=headers, timeout=self.http_timeout)
+        except requests.ConnectionError as e:
+            commons.print_msg(Jira.clazz, method, "Failed retrieving story detail from call to {}".format(
+                story_detail.get('url', '')), 'ERROR')
+            commons.print_msg(Jira.clazz, method, e, 'ERROR')
+            exit(1)
+        
+        if resp.status_code == 200:
+            json_data = json.loads(resp.text)
+            commons.print_msg(Jira.clazz, method, json_data)
+        else:
+            commons.print_msg(Jira.clazz, method, "Failed retrieving story detail from call to {url}. \r\n "
+                                                         "Response: {response}".format(url=story_detail.get('url', ''),
+                                                                                       response=resp.text), 'WARN')
+
+        commons.print_msg(Jira.clazz, method, 'end')
+        return json_data
+
+    def tag_stories_in_commit(self, story_list):
+        method = 'tag_stories_in_commit'
+        commons.print_msg(Jira.clazz, method, 'begin')
+
+        for story in story_list:
+            version = '{0}-{1}'.format(self.config.project_name, self.config.version_number)
+            self._add_version_to_project(version)
+            self._add_version_to_story(story, version)
+
+        commons.print_msg(Jira.clazz, method, 'end')
+
+    def _add_version_to_project(self, version):
```

Comment:
This seems to execute for each identified labeled story, which results in the same version attempting to be created multiple times in situations where there are multiple stories referenced in the commits.

Reply:
Added a check to see if the version exists on the project and skip version creation if it does.